pytorch_lightning/overrides/base.py

```diff
@@ -46,6 +46,13 @@ def forward(self, *inputs, **kwargs):
 
         if running_stage == RunningStage.TRAINING:
             output = self.module.training_step(*inputs, **kwargs)
+
+            # In manual_optimization, we need to prevent DDP reducer as
+            # it is done manually in ``LightningModule.manual_backward``
+            # `require_backward_grad_sync` will be reset
+            # ddp_plugin ``post_training_step`` hook
+            if not self.module.automatic_optimization:
+                self.module.trainer.model.require_backward_grad_sync = False
```

Comment:
I think this should not be here, looking at the call: `self.module.trainer.model` is the pytorch DDP wrapper.
So this can be added outside, maybe in the DDP training type plugin

Reply:
nvm, although this looks suboptimal, @justusschock explained why it is necessary.